src/pricing-helper.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  Address,
+  LoggerConstructor,
+  Logger,
+  Token,
+  PoolPrices,
+  ExchangePrices,
+} from './types';
+import {
+  SwapSide,
+  SETUP_RETRY_TIMEOUT,
+  FETCH_POOL_INDENTIFIER_TIMEOUT,
+  FETCH_POOL_PRICES_TIMEOUT,
+} from './constants';
+import { DexAdapterService } from './dex';
+
+export class PricingHelper {
+  logger: Logger;
+
+  constructor(
+    protected dexAdapterService: DexAdapterService,
+    loggerConstructor: LoggerConstructor,
+  ) {
+    this.logger = loggerConstructor(
+      `PricingHelper_${dexAdapterService.network}`,
+    );
+  }
+
+  private async initializeDex(dexKey: string, blockNumber: number) {
+    try {
+      const dexInstance = this.dexAdapterService.getDexByKey(dexKey);
+
+      if (!dexInstance.initializePricing) return;
+
+      return await dexInstance.initializePricing(blockNumber);
+    } catch (e) {
+      this.logger.error('Error_startListening:', e);
+      setTimeout(
+        () => this.initializeDex(dexKey, blockNumber),
+        SETUP_RETRY_TIMEOUT,
+      );
+    }
+  }
+
+  public getAllDexKeys(): string[] {
+    return this.dexAdapterService.getAllDexKeys();
+  }
+
+  public async initialize(blockNumber: number, dexKeys: string[]) {
+    return await Promise.all(
+      dexKeys.map(key => this.initializeDex(key, blockNumber)),
+    );
+  }
+
+  public async getPoolIdentifiers(
+    from: Token,
+    to: Token,
+    side: SwapSide,
+    blockNumber: number,
+    dexKeys: string[],
+    filterConstantPricePool: boolean = false,
+  ): Promise<(string[] | null)[]> {
```

Comment:
Should return a mapping of dex keys to arrays of strings rather than array of arrays, so that it matches what is expected by `getPoolPrices`.

Reply:
fixed